jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/TreeSingleSourcePathsImpl.java

```diff
@@ -135,7 +135,7 @@ public double getWeight(V targetVertex)
         }
 
         double weight = 0d;
-        while (p != null && !p.equals(source)) {
+        while (p != null && !cur.equals(source)) {
```

Comment:
I don't think we need this 2nd check at all. I think we can just write `while(p!=null)`. Can you verify this? This method recreates the path by walking back from a node to the source by following a trail of edges. As soon as the source is reached, map.get(source) will return null. Better to check this though.

Reply:
Can you please verify this? Thnx!